routing/datasource.go

```diff
@@ -241,6 +250,87 @@ func isTreePredicate(name string) bool {
 	}
 }
 
+func getFreeStringArg(count int, p *eskip.Predicate) ([]string, error) {
+	err := func() error { return fmt.Errorf("invalid predicate args in %s", p.Name) }
```

Comment:
I don't like that, because it does not show exactly what was wrong.
Can you please have different error messages for length and type?